Firestore/Example/Tests/Integration/API/FIRDatabaseTests.mm

```diff
@@ -256,6 +256,157 @@ - (void)testMergeReplacesArrays {
   XCTAssertEqualObjects(document.data, finalData);
 }
 
+- (void)testCannotSpecifyFieldMaskForMissingField {
+  FIRDocumentReference *doc = [[self.db collectionWithPath:@"rooms"] documentWithAutoID];
+
+  XCTAssertThrowsSpecific(
+      {
+        [doc setData:@{} mergeFields:@[ @"foo" ]];
+      },
+      NSException,
+      @"Field 'foo' is specified in your field mask but missing from your input data.");
+}
+
+- (void)testCanSetASubsetOfFieldsUsingMask {
+  FIRDocumentReference *doc = [[self.db collectionWithPath:@"rooms"] documentWithAutoID];
+
+  NSDictionary<NSString *, id> *initialData =
+      @{ @"des" : @"Description",
+         @"owner" : @{@"name" : @"Jonny", @"email" : @"abc@xyz.com"} };
+
+  NSDictionary<NSString *, id> *finalData = @{@"des" : @"Description", @"owner" : @"Sebastian"};
+
+  [self writeDocumentRef:doc data:initialData];
+
+  XCTestExpectation *completed =
+      [self expectationWithDescription:@"testCanSetASubsetOfFieldsUsingMask"];
+
+  [doc setData:@{@"desc" : @"NewDescription", @"owner" : @"Sebastian"}
```

Comment:
The rest of the test uses `"des"`. They should probably all be `"desc"`?

Looks like all of the tests do this.

Reply:
Update everything to "desc". Thanks for finding this.